airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/SerializedBufferingStrategy.java

```diff
@@ -77,18 +89,18 @@ public boolean addRecord(final AirbyteStreamNameNamespacePair stream, final Airb
     }
     final long actualMessageSizeInBytes = streamBuffer.accept(message.getRecord());
     totalBufferSizeInBytes += actualMessageSizeInBytes;
+    // Flushes buffer when either the buffer was completely filled or only a single stream was filled
```

Comment:
Nonblocking: comment above conditional could be changed to well named helper method with docs, i.e. 
```
if (shouldFlushAll()) {
...
```